### PR TITLE
Fix `Layout/LineLength` cop failure in case of tabbed YARD-comment-like string

### PR DESCRIPTION
**Pull request by viralpraxis**
The following ruby code leads to an error:

```
rubocop -c <(echo "Layout/LineLength:\n AllowURI: true\n Max: 1") --stdin bug.rb --only Layout/LineLength -d <<EOF
  expect(some_exception_variable) {|e| e.url.should == 'http://host/path'}
EOF
```

(there's a `\t` in the beginning)

Backtrace:

```
An error occurred while Layout/LineLength cop was inspecting bug.rb.
undefined method `offset' for nil
lib/rubocop/cop/mixin/line_length_help.rb:76:in `extend_uri_end_position'
lib/rubocop/cop/mixin/line_length_help.rb:44:in `find_excessive_uri_range'
lib/rubocop/cop/layout/line_length.rb:359:in `check_uri_line'
lib/rubocop/cop/layout/line_length.rb:258:in `check_line'
lib/rubocop/cop/layout/line_length.rb:103:in `block in on_investigation_end'
lib/rubocop/cop/layout/line_length.rb:102:in `each'
lib/rubocop/cop/layout/line_length.rb:102:in `each_with_index'
lib/rubocop/cop/layout/line_length.rb:102:in `on_investigation_end'
```

-----------------

Before submitting the PR make sure the following are checked:

* [X] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [X] Wrote [good commit messages][1].
* [X] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/mixin/line_length_help.rb [hunk 1]
```diff
@@ -37,12 +37,13 @@ def find_excessive_uri_range(line)
...
 
-        begin_position, end_position = last_uri_match.offset(0).map do |pos|
-          pos + indentation_difference(line)
-        end
-
+        begin_position, end_position = last_uri_match.offset(0)
         end_position = extend_uri_end_position(line, end_position)
 
+        line_indentation_difference = indentation_difference(line)
+        begin_position += line_indentation_difference
```
Comment (koic) [inline lib/rubocop/cop/mixin/line_length_help.rb:44]: Is this line of processing necessary?

Comment (viralpraxis) [reply lib/rubocop/cop/mixin/line_length_help.rb:44]: It does make a difference in some cases.

For example, this line `"\t\t# See additional infor at http://x.com/rr"` with config

```
'Layout/LineLength' => { 'Max' => 30, 'AllowURI' => true }
'Layout/IndentationWidth' => { 'Width' => 2 },
```

would be marked as an offense (`Line is too long. [45/30]`) if we keep `begin_position += line_indentation_difference` (otherwise it wouldn't)

In general I believe this is a relatively rare edge case which does not make much difference in practice.

Comment (koic) [reply lib/rubocop/cop/mixin/line_length_help.rb:44]: It seems that no failing tests exist even if this line is removed. Could you add a new test code for this?

Comment (viralpraxis) [reply lib/rubocop/cop/mixin/line_length_help.rb:44]: Sure; I added a test based on my comment above.

**Comment by koic** [comment]
Thanks!
